x-pack/plugins/apm/public/components/shared/kuery_bar/index.tsx

```diff
@@ -95,22 +102,22 @@ export function KueryBar(props: { prepend?: React.ReactNode | string }) {
         (await data.autocomplete.getQuerySuggestions({
           language: 'kuery',
           indexPatterns: [indexPattern],
-          boolFilter: getBoolFilter({
-            groupId,
-            processorEvent,
-            serviceName,
-            environment,
-            urlParams,
-          }),
+          boolFilter:
+            props.boolFilter ??
+            getBoolFilter({
+              groupId,
+              processorEvent,
+              serviceName,
+              environment,
+              urlParams,
+            }),
```

Comment:
Can we get rid of `getBoolFilter` now and supply the filters explicitly like you do for the backends route?

A route can take `searchBarOptions` so even if the same instance of `<SearchBar />` is used we should be able to adapt the placeholder and filter per route:
https://github.com/elastic/kibana/blob/3ab852f213f76a669e73e27ab4f7b4cd622e6658/x-pack/plugins/apm/public/components/routing/service_detail/index.tsx#L105-L108

```suggestion

```